singer-io/tap-saasoptics
# Add deleted stream for invoices, contracts, and transactions

# Description of change
SaaS Optics provides deleted records from a `deleted` endpoint. This PR adds a stream per-object to request and bookmark deleted objects.

The filtering on this endpoint does not appear to work, so the tap will currently paginate through all deleted records and only emit newer ones. This is the same behavior as other datetime-oriented streams for this tap.

# Manual QA steps
 - Ran through queries with Postman to confirm behavior
 - Ran through a local sync to confirm bookmarking works as expected
 
# Risks
 - Low, any large backwards-incompatible issues should be rooted out by running test syncs.
 
# Rollback steps
 - revert this branch, bump patch version
